crates/adapters/src/adhoc/format.rs

```diff
@@ -0,0 +1,45 @@
+use arrow::array::RecordBatch;
+use arrow::error::ArrowError;
+use arrow::util::display::{ArrayFormatter, FormatOptions};
+use comfy_table::{Cell, Table};
+pub(crate) fn create_table(results: &[RecordBatch]) -> Result<Table, ArrowError> {
+    let options = FormatOptions::default().with_display_error(true);
+    let mut table = Table::new();
+    table.load_preset("||--+-++|    ++++++");
+
+    if results.is_empty() {
+        return Ok(table);
+    }
+
+    let schema = results[0].schema();
+
+    let mut header = Vec::new();
+    for field in schema.fields() {
+        header.push(Cell::new(field.name()));
+    }
+    table.set_header(header);
+
+    for batch in results {
+        let formatters = batch
```

Comment:
is this needed for each batch or can it be done just once?

Reply:
all our batches are the same so could be done once but this is also what the orignal code did in the arrow library where I took this function from